File: budgetMicroservice/src/main/java/com/budgetMicroservice/dto/SortDTO.java
package com.budgetMicroservice.dto;

import com.budgetMicroservice.enumerator.SortDirection;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class SortDTO {
    private boolean sorted;
    private boolean unsorted;
    private boolean empty;
    private String field;
    private SortDirection direction;
}

File: client/src/app/services/models/sort-dto.ts
/**
 * Data Transfer Object representing sorting options, including sorting direction and field for a query.
 */
export type SortDto = ({

/**
 * Indicates whether the data is sorted.
 */
'sorted'?: boolean;

/**
 * Indicates whether the data is unsorted.
 */
'unsorted'?: boolean;

/**
 * Indicates whether the data is empty (i.e., no sorting applied).
 */
'empty'?: boolean;

/**
 * The field by which the data should be sorted.
 */
'field': string;

/**
 * The direction of sorting (ASC or DESC).
 */
'direction': 'ASC' | 'DESC';
}) | null;
